### ISSUE TITLE
Invalid block headers when laptop wakes up and bitcoind announces block headers

### ISSUE THREAD

**Issue by Christewart** [bitcoin-s/bitcoin-s#5072]
Seen on on c20c7cba

```
2023-05-15 17:36:23,375UTC WARN [HikariPool] bitcoin-s-chain-db - Thread starvation or clock leap detected (housekeeper delta=15m29s305ms).
2023-05-15 17:36:24,436UTC WARN [HikariPool] bitcoin-s-node-db - Thread starvation or clock leap detected (housekeeper delta=15m29s305ms).
2023-05-15 17:36:24,502UTC WARN [HikariPool] bitcoin-s-wallet-db - Thread starvation or clock leap detected (housekeeper delta=15m29s306ms).
2023-05-15 17:36:24,529UTC WARN [HikariPool] bitcoin-s.dlc.db - Thread starvation or clock leap detected (housekeeper delta=15m29s306ms).
2023-05-15 18:52:33,085UTC WARN [HikariPool] bitcoin-s-chain-db - Thread starvation or clock leap detected (housekeeper delta=1h16m9s711ms).
2023-05-15 18:52:34,151UTC WARN [HikariPool] bitcoin-s-node-db - Thread starvation or clock leap detected (housekeeper delta=1h16m9s716ms).
2023-05-15 18:52:34,268UTC WARN [HikariPool] bitcoin-s-wallet-db - Thread starvation or clock leap detected (housekeeper delta=1h16m9s766ms).
2023-05-15 18:52:34,301UTC WARN [HikariPool] bitcoin-s.dlc.db - Thread starvation or clock leap detected (housekeeper delta=1h16m9s772ms).
2023-05-15 18:52:48,653UTC INFO [DataMessageHandler] Received block message with hash 00000000000000000005a48352f43ff8f8dcba15f1574991ed07797162848f3b
2023-05-15 18:52:48,655UTC INFO [DataMessageHandler] Received headers message with 1 headers from peer=Peer(127.0.0.1:8333) state=DoneSyncing
2023-05-15 18:52:48,684UTC INFO [ChainHandler] Processed headers from height=789886 to 789887. Best hash=00000000000000000005a48352f43ff8f8dcba15f1574991ed07797162848f3b
2023-05-15 18:52:48,684UTC INFO [PeerManager] Now syncing filter headers from Peer(127.0.0.1:8333) in state=HeaderSync(Peer(127.0.0.1:8333))
2023-05-15 18:52:48,958UTC INFO [DataMessageHandler] Ignoring inv message=InventoryMessage(1 inv(s), Inventory(MsgBlock, blockHashBE=DoubleSha256BDigestBE(00000000000000000000f3502c3efe3e64b5fa151f58f04bba74f1e8ab8108ec))) while in state=FilterHeaderSync(Peer(127.0.0.1:8333))
2023-05-15 18:52:48,959UTC INFO [DLCWallet$DLCWalletImpl] Processing block=00000000000000000005a48352f43ff8f8dcba15f1574991ed07797162848f3b heightOpt=Some(789887)
2023-05-15 18:52:48,962UTC INFO [DLCWallet$DLCWalletImpl] Finished processing of block=00000000000000000005a48352f43ff8f8dcba15f1574991ed07797162848f3b. It took 5ms
2023-05-15 18:52:49,632UTC INFO [ChainHandler] Processed filters headers from height=789887 to 789887. Best filterheader.blockHash=00000000000000000005a48352f43ff8f8dcba15f1574991ed07797162848f3b
2023-05-15 18:52:51,080UTC INFO [DataMessageHandler] Beginning to sync filters from startHeight=789886
2023-05-15 18:52:51,113UTC INFO [PeerMessageSender] Requesting compact filters from FilterSyncMarker(startHeight=789887, stopBlockHash=00000000000000000005a48352f43ff8f8dcba15f1574991ed07797162848f3b)
2023-05-15 18:52:51,113UTC INFO [DataMessageHandler] Ignoring inv message=InventoryMessage(1 inv(s), Inventory(MsgBlock, blockHashBE=DoubleSha256BDigestBE(00000000000000000000fb9d978a9ea2495924cc0428f324c55502d22b4cc494))) while in state=FilterSync(Peer(127.0.0.1:8333))
2023-05-15 18:52:51,114UTC INFO [DataMessageHandler] Ignoring inv message=InventoryMessage(1 inv(s), Inventory(MsgBlock, blockHashBE=DoubleSha256BDigestBE(00000000000000000000e9a3e863e98e760219abb11998e8e5bea979541976f9))) while in state=FilterSync(Peer(127.0.0.1:8333))
2023-05-15 18:52:51,329UTC INFO [DataMessageHandler] Processing 1 filters
2023-05-15 18:52:51,333UTC INFO [ChainHandler] Processed filters from height=789887 to 789887. Best filter.blockHash=00000000000000000005a48352f43ff8f8dcba15f1574991ed07797162848f3b
2023-05-15 18:52:51,334UTC INFO [DLCWallet$DLCWalletImpl] Processing 1 block filters for 0 utxos and 0 scripts
2023-05-15 18:52:51,344UTC INFO [DataMessageHandler] We are synced
2023-05-15 18:52:53,156UTC INFO [DataMessageHandler] Received block message with hash 0000000000000000000188404d6afa76c337db71abd61fbbd750910cc2b44028
2023-05-15 18:52:53,166UTC INFO [DataMessageHandler] Received headers message with 1 headers from peer=Peer(127.0.0.1:8333) state=DoneSyncing
2023-05-15 18:52:53,223UTC WARN [DataMessageHandler] Invalid headers of count CompactSizeUInt(1) sent from Peer(127.0.0.1:8333) in state=DoneSyncing
2023-05-15 18:52:53,227UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=0000000000000000000188404d6afa76c337db71abd61fbbd750910cc2b44028, skipping processing in wallet
2023-05-15 18:52:53,227UTC INFO [DLCWallet$DLCWalletImpl] Processing block=0000000000000000000188404d6afa76c337db71abd61fbbd750910cc2b44028 heightOpt=None
2023-05-15 18:52:53,382UTC INFO [DataMessageHandler] Received block message with hash 000000000000000000034407b44c846ff2b19753c98a1efa13cccad03072ba18
2023-05-15 18:52:53,384UTC INFO [DataMessageHandler] Received headers message with 1 headers from peer=Peer(127.0.0.1:8333) state=DoneSyncing
2023-05-15 18:52:53,408UTC WARN [DataMessageHandler] Invalid headers of count CompactSizeUInt(1) sent from Peer(127.0.0.1:8333) in state=DoneSyncing
2023-05-15 18:52:53,410UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=000000000000000000034407b44c846ff2b19753c98a1efa13cccad03072ba18, skipping processing in wallet
2023-05-15 18:52:53,410UTC INFO [DLCWallet$DLCWalletImpl] Processing block=000000000000000000034407b44c846ff2b19753c98a1efa13cccad03072ba18 heightOpt=None
2023-05-15 18:52:53,564UTC INFO [DataMessageHandler] Received block message with hash 000000000000000000019cd5c76e09be0690bda2ecfa487f2e3f63e74f6ad6a7
2023-05-15 18:52:53,567UTC INFO [DataMessageHandler] Received headers message with 1 headers from peer=Peer(127.0.0.1:8333) state=DoneSyncing
2023-05-15 18:52:53,594UTC WARN [DataMessageHandler] Invalid headers of count CompactSizeUInt(1) sent from Peer(127.0.0.1:8333) in state=DoneSyncing
2023-05-15 18:52:53,595UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=000000000000000000019cd5c76e09be0690bda2ecfa487f2e3f63e74f6ad6a7, skipping processing in wallet
2023-05-15 18:52:53,595UTC INFO [DLCWallet$DLCWalletImpl] Processing block=000000000000000000019cd5c76e09be0690bda2ecfa487f2e3f63e74f6ad6a7 heightOpt=None
2023-05-15 18:53:37,122UTC INFO [BitcoinSServerMain$] @@@@@@@@@@@@@@@@@@@@@ Shutting down BitcoinSServerMain$ @@@@@@@@@@@@@@@@@@@@@
2023-05-15 18:53:37,122UTC ERROR [BitcoinSServerMain] Exiting process
2023-05-15 18:53:37,129UTC INFO [NodeCallbackStreamManager] Done draining akka streams for NodeCallbackStreamManager, it took=7ms
2023-05-15 18:53:37,130UTC INFO [HikariDataSource] bitcoin-s-node-db - Shutdown initiated...
2023-05-15 18:53:37,132UTC INFO [HikariDataSource] bitcoin-s-node-db - Shutdown completed.
2023-05-15 18:53:37,134UTC INFO [HikariDataSource] bitcoin-s.dlc.db - Starting...
2023-05-15 18:53:37,135UTC ERROR [PoolBase] bitcoin-s.dlc.db - JMX name (bitcoin-s.dlc.db) is already registered.
2023-05-15 18:53:37,135UTC INFO [HikariDataSource] bitcoin-s.dlc.db - Start completed.
2023-05-15 18:53:37,135UTC INFO [HikariDataSource] bitcoin-s.dlc.db - Shutdown initiated...
2023-05-15 18:53:37,135UTC INFO [HikariDataSource] bitcoin-s.dlc.db - Shutdown completed.
2023-05-15 18:53:37,136UTC INFO [HikariDataSource] bitcoin-s-chain-db - Shutdown initiated...
2023-05-15 18:53:37,141UTC INFO [HikariDataSource] bitcoin-s-chain-db - Shutdown completed.
2023-05-15 18:53:37,145UTC INFO [TorClient] No process found, binary wasn't started!
2023-05-15 18:53:37,147UTC INFO [WalletAppConfig] Stopping wallet rebroadcast task
2023-05-15 18:53:37,147UTC INFO [HikariDataSource] bitcoin-s-wallet-db - Shutdown initiated...
2023-05-15 18:53:37,148UTC INFO [HikariDataSource] bitcoin-s-wallet-db - Shutdown completed.
2023-05-15 18:53:37,148UTC INFO [HikariDataSource] bitcoin-s.dlc.db - Shutdown initiated...
2023-05-15 18:53:37,149UTC INFO [HikariDataSource] bitcoin-s.dlc.db - Shutdown completed.
2023-05-15 18:53:37,155UTC INFO [BitcoinSServerMain] Stopped neutrino node
```

**Comment by Christewart** [comment]
Related to #5154 , this might not be totally fixed in #5154 

This is a unique case since the connection wasn't actually dropped and then reconnected. It was one continuous connection.

**Comment by Christewart** [comment]
Fixed by a combination of #5154 and #5201
